src/main/main.c

```diff
@@ -425,8 +425,11 @@ void init(void)
     flashLedsAndBeep();
 
 #ifdef MAG
-    if (sensors(SENSOR_MAG))
-        compassInit();
+    if (sensors(SENSOR_MAG)) {
+        if (!compassInit()) {
```

Comment:
I think this is not where it belongs. All other sensors are initialised inside `sensorsAutodetect()` call. Worth thinking about cleaning this up.